File: 2021/src/test/java/de/jfriemel/aoc/days/Day11Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day11Test {

    private static final String INPUT_STRING = """
            5483143223
            2745854711
            5264556173
            6141336146
            6357385478
            4167524645
            2176841721
            6882881134
            4846848554
            5283751526""";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day11().part1(INPUT);
        assertEquals("1656", result);
    }

    @Test
    public void testPart2() {
        String result = new Day11().part2(INPUT);
        assertEquals("195", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day11Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day11Test {
    private val input = """
        ...#......
        .......#..
        #.........
        ..........
        ......#...
        .#........
        .........#
        ..........
        .......#..
        #...#.....
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day11.part1(input)
        Assertions.assertEquals("374", result)
    }

    @Test
    fun testPart2() {
        val result = Day11.part2(input)
        Assertions.assertEquals("82000210", result)
    }
}
